src/sentry_logs.c

```diff
@@ -4,9 +4,237 @@
+
+    // Wait for all adding operations to complete using condition variable
+    sentry__mutex_lock(&g_logs_single_state.adding_lock);
+    while (sentry__atomic_fetch(&g_logs_single_state.queue.adding) > 0) {
+        sentry__cond_wait(
+            &g_logs_single_state.adding_done, &g_logs_single_state.adding_lock);
+    }
+    sentry__mutex_unlock(&g_logs_single_state.adding_lock);
+
+    sentry_value_t logs = sentry_value_new_object();
+    sentry_value_t log_items = sentry_value_new_list();
+    int i;
+    long queue_len = sentry__atomic_fetch(&g_logs_single_state.queue.index);
+    for (i = 0; i < queue_len; i++) {
+        sentry_value_append(log_items, g_logs_single_state.queue.logs[i]);
+    }
+    sentry_value_set_by_key(logs, "items", log_items);
+    // sending of the envelope
+    sentry_envelope_t *envelope = sentry__envelope_new();
+    sentry__envelope_add_logs(envelope, logs);
+    SENTRY_WITH_OPTIONS (options) {
+        sentry__capture_envelope(options->transport, envelope);
+    }
+    sentry__atomic_store(&g_logs_single_state.queue.index, 0);
+    sentry__atomic_fetch_and_add(&g_logs_single_state.flushing, -1);
+    SENTRY_DEBUGF("Time to flush %i items is %llu us\n", i,
+        sentry__usec_time() - before_flush);
+}
+
+static void
+setup_bgworker(void)
+{
+    g_logs_single_state.timer_worker = sentry__bgworker_new(NULL, NULL);
+    if (g_logs_single_state.timer_worker) {
+        sentry__bgworker_setname(
+            g_logs_single_state.timer_worker, "sentry-logs-timer");
+        if (sentry__bgworker_start(g_logs_single_state.timer_worker) != 0) {
+            SENTRY_WARN("Failed to start timer bgworker");
+            sentry__bgworker_decref(g_logs_single_state.timer_worker);
+            g_logs_single_state.timer_worker = NULL;
+            sentry__atomic_fetch_and_add(
+                &g_logs_single_state.timer_running, -1);
+        }
+    } else {
+        SENTRY_WARN("Failed to create timer bgworker");
+        sentry__atomic_fetch_and_add(&g_logs_single_state.timer_running, -1);
+    }
+}
+static void
+generate_timer_task(void)
+{
+    // Submit timer task if worker is available and no task is
+    // already submitted
+    if (g_logs_single_state.timer_worker
+        && !sentry__atomic_fetch(&g_logs_single_state.timer_task_submitted)) {
+        sentry__atomic_store(&g_logs_single_state.timer_task_submitted, 1);
+        if (sentry__bgworker_submit(
+                g_logs_single_state.timer_worker, timer_task_func, NULL, NULL)
+            != 0) {
+            SENTRY_WARN("Failed to submit timer task");
+            sentry__atomic_fetch_and_add(
+                &g_logs_single_state.timer_running, -1);
+            sentry__atomic_store(&g_logs_single_state.timer_task_submitted, 0);
+        }
+    } else {
+        // Worker unavailable or task already submitted,
+        // reset the running flag
+        sentry__atomic_fetch_and_add(&g_logs_single_state.timer_running, -1);
+    }
+}
```

Comment:
the same as above, the `timer_task` is not implemented as a "task" but as a thread-loop. This requires a `sentry__thread_spawn`, after the above did the `sentry__thread_new`.